README.md

```diff
@@ -8,233 +8,156 @@
-output = Vacuum(4) >> X8
-
-# lossy X8
-noise = lambda: np.random.uniform(size=4)
-X8_noisy = (Sgate(r=0.9 + 0.1*noise(), phi=0.1*noise())
-                >> Attenuator(0.89 + 0.01*noise())
-                >> Interferometer(4)
-                >> Attenuator(0.95 + 0.01*noise())
-               )
-
-# 2-mode Bloch Messiah decomposition
-bloch_messiah = Sgate(r=[0.1,0.2]) >> BSgate(theta=-0.1, phi=2.1) >> Dgate(x=[0.1, -0.4])
-my_state = Vacuum(2) >> bloch_messiah
-```
+States can be measured via contraction with a dual state.
 
-## 4. Measurements
-In order to perform a measurement, we use the left-shift operator, e.g. `coh << sq` (think of the left-shift on a state as "closing" the circuit).
 ```python
-leftover = Vacuum(4) >> X8 << SqueezedVacuum(r=10.0, phi=np.pi)[2]  # a homodyne measurement of p=0.0 on mode 2
+cat >> Number(mode=0, n=3).dual # equivalent to a PNR measurement of 4 photons
 ```
 
-Transformations can also be applied in the dual sense by using the left-shift operator `<<`:
-```python
-Attenuator(0.5) << Coherent(0.1, 0.2) == Coherent(0.1, 0.2) >> Amplifier(2.0)
-```
-This has the advantage of modelling lossy detectors without applying the loss channel to the state going into the detector, which can be overall faster e.g. if the state is kept pure by doing so.
 
-## 5. Detectors
-There are two types of detectors in Mr Mustard. Fock detectors (PNRDetector and ThresholdDetector) and Gaussian detectors (Homodyne, Heterodyne, Generaldyne).
+States can also generate samples via the built-in sampler.
 
-The PNR and Threshold detectors return an array of unnormalized measurement results, meaning that the elements of the array are the density matrices of the leftover systems, conditioned on the outcomes:
 ```python
-results = Gaussian(2) << PNRDetector(efficiency = 0.9, modes = [0])
-results[0]  # unnormalized dm of mode 1 conditioned on measuring 0 in mode 0
-results[1]  # unnormalized dm of mode 1 conditioned on measuring 1 in mode 0
-results[2]  # unnormalized dm of mode 1 conditioned on measuring 2 in mode 0
-# etc...
-```
-The trace of the leftover density matrices will yield the success probability. If multiple modes are measured then there is a corresponding number of indices:
-```python
-results = Gaussian(3) << PNRDetector(efficiency = [0.9, 0.8], modes = [0,1])
-results[2,3]  # unnormalized dm of mode 2 conditioned on measuring 2 in mode 0 and 3 in mode 1
-# etc...
-```
-Set a lower `settings.PNR_INTERNAL_CUTOFF` (default 50) to speed-up computations of the PNR output.
+pnr = PNRSampler(cutoff=100)
+pnr_samples = pnr.sample(state=cat, n_samples=100, seed=None)
 
-## 6. Comparison operator `==`
-States support the comparison operator:
-```python
->>> bunched = (Coherent(1.0) & Coherent(1.0)) >> BSgate(np.pi/4)
->>> bunched.get_modes(1) == Coherent(np.sqrt(2.0))
-True
-```
-As well as transformations (gates and circuits):
-```python
->>> Dgate(np.sqrt(2)) >> Attenuator(0.5) == Attenuator(0.5) >> Dgate(1.0)
-True
```

Comment:
Can we also have keep some `>>>` lines here, such that they may be ran with `doctest`? Doing so can help ensure that this file does not fall out of date as easily going forward.